elan/rpc/eclient.go

```diff
@@ -108,9 +108,16 @@ func (e *elanClient) uploadOne(entry *uploadinfo.Entry) error {
 	if err != nil {
 		return err
 	}
-	defer wr.Close()
+
+	defer func() {
+		if closeErr := wr.Close(); closeErr != nil {
```

Comment:
I was thinking the other way around, like
```
if closeErr := wr.Close(); err == nil {
    err = closeErr
}
```
If `err` is already not nil then I suspect replacing it with `closeErr` isn't going to make things easier to understand.

Reply:
Ah of course! Thanks